pyspecProcScripts/integral_w_error.py

```diff
@@ -72,46 +71,25 @@ def integral_w_errors(
             "You have extra (non-phase cycling, non-indirect) dimensions: "
             + str(extra_dims)
         )
-    collected_variance = psp.ndshape(
-        [psp.ndshape(s)[indirect], len(error_path)], [indirect, "pathways"]
-    ).alloc()
-    avg_error = []
-    for j in range(len(error_path)):
-        # calculate N₂ Δf² σ², which is the variance of the integral
-        # (by error propagation) where N₂ is the number of points in
-        # the indirect dimension
-        s_forerror = select_pathway(s, error_path[j])
-        # previous line wipes everything out and starts over -- why not use
-        # collected_variance above, as I had originally set up --> part of
-        # issue #44
-        if j == 0:
-            N2 = psp.ndshape(s_forerror)[direct]
-        # mean divides by N₁ (indirect), integrate multiplies by Δf, and the
-        # mean sums all elements (there are N₁N₂ elements)
-        s_forerror -= s_forerror.C.mean_all_but([indirect, direct]).mean(
-            direct
-        )
-        s_forerror.run(lambda x: abs(x) ** 2 / 2).mean_all_but(
-            [direct, indirect]
-        ).mean(direct)
-        s_forerror *= df**2  # Δf
-        s_forerror *= N2
-        avg_error.append(s_forerror)
-    avg_error = sum(avg_error) / len(avg_error)
-    # {{{ variance calculation for debug
-    variance_debug = False
-    if variance_debug:
-        print(
-            "(inside automatic routine) the stdev seems to be",
-            np.sqrt(collected_variance / (df * N2)),
-        )
-        print(
-            "automatically calculated integral error:",
-            np.sqrt(collected_variance.data),
-        )
-    # }}}
-    s = select_pathway(s, sig_path)
-    retval = s.integrate(direct).set_error(psp.sqrt(s_forerror.data))
+    # Determine smaller slice within signal bounds where ph cycling noise most
+    # likely resides
+    frq_filter_push = (frq_slice[1] - frq_slice[0]) / 3
+    frq_filter_bounds = (
+        frq_slice[0] + frq_filter_push,
+        frq_slice[-1] - frq_filter_push,
+    )
+    variance = calc_masked_error(
+        s,
+        frq_filter_bounds,
+        sig_path,
+        indirect=indirect,
+        excluded_pathways=excluded_pathways,
+        fl=fl,
+    )
```

Comment:
this needs to be moved up, before you take `frq_slice` to calculate the integral.

Reply:
moved before frequency slice